spinn_front_end_common/interface/interface_functions/front_end_common_insert_edges_to_live_packet_gatherers.py

```diff
@@ -0,0 +1,179 @@
+from pacman.model.graphs.application import ApplicationVertex, ApplicationEdge
+from pacman.model.graphs.machine import MachineEdge
+from spinn_front_end_common.utilities import exceptions
+from spinn_utilities.progress_bar import ProgressBar
+
+
+class FrontEndCommonInsertEdgesToLivePacketGatherers(object):
+    """ adds edges from the recorded vertices to the local LPGs
+
+    """
+
+    def __call__(
+            self, live_packet_gatherers, placements,
+            live_packet_recorder_recorded_vertex_type,
+            live_packet_gatherers_to_vertex_mapping, machine,
+            machine_graph, application_graph=None, graph_mapper=None):
+        """ adds edges from the recorded vertices to the local LPGs
+
+        :param live_packet_gatherers: the set of params and vertices to link to
+        :param placements: the placements object
+        :param live_packet_gatherers_to_vertex_mapping: the mapping of LPG
+         parameters and the machine vertex associated with it
+        :param machine: the SpiNNaker machine
+        :param machine_graph: the machine graph
+        :param application_graph:  the app graph
+        :param graph_mapper: the graph mapper between app and machine graph
+        :rtype: None
+        """
+
+        progress_bar = ProgressBar(
+            total_number_of_things_to_do=len(live_packet_gatherers),
+            string_describing_what_being_progressed=(
+                "Adding edges to the machine graph between the vertices to "
+                "which live output has been requested and its local Live "
+                "Packet Gatherer"))
+
+        for live_packet_gatherers_param in live_packet_gatherers:
```

Comment:
Iterate over `progress_bar.over(live_packet_gatherers)` here so that you don't need to have `progress_bar.update()` and `progress_bar.end()` at the end of the loop (lines 85–86).